fs_storage/models/fs_storage.py

```diff
@@ -162,6 +162,13 @@ def __init__(self, env, ids=(), prefetch_ids=()):
     def _server_env_fields(self):
         return {"protocol": {}, "options": {}, "directory_path": {}}
 
+    def _server_env_read_from_config(self, field_name, config_getter):
+        value = super()._server_env_read_from_config(field_name, config_getter)
+        # replace {db_name} with the dbname
+        if field_name == "directory_path":
+            value = value.format(db_name=self.env.cr.dbname)
+        return value
```

Comment:
This is also probably not the best place to implement this, as if we do this, it should not behave differently when the value is entered manually or via server environment.